From 47b2e0039cb2ca58282f94e56ff851c149467b52 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Tue, 30 Apr 2013 14:15:17 +0800
Subject: [PORT FROM MAIN] Awesomeplayer: adjust the seek for audio of mkv container

BZ: 48604

Original patch:
R3: 30123(BZ: 18523)
R2: 15390(BZ: 6907)

The seek of audio in mkv container is not correct, for every audio frame
is keyframe. Also when the video seek to the end, the audio should also to
be the end, so the media will be complete.

Change-Id: Ieafa627e7b8ca8887f39c29077e3d4f7393561cb
Orig-Change-Id: Ifdfa41dee9fb5bf04356317222f9fbcb7dbf04f0
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index cd26d10..f3c0908 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1679,6 +1679,7 @@ void AwesomePlayer::onVideoEvent() {
                 // So video playback is complete, but we may still have
                 // a seek request pending that needs to be applied
                 // to the audio track.
+                SeekType seekTemp = mSeeking;
                 if (mSeeking != NO_SEEK) {
                     ALOGV("video stream ended while seeking!");
                 }
@@ -1689,6 +1690,11 @@ void AwesomePlayer::onVideoEvent() {
                     startAudioPlayer_l();
                 }
 
+                if (seekTemp != NO_SEEK) {
+                    modifyFlags(AUDIO_AT_EOS, SET); // video is eos, end the audio?
+                    mVideoTimeUs = mSeekTimeUs;
+                }
+
                 modifyFlags(VIDEO_AT_EOS, SET);
                 postStreamDoneEvent_l(err);
                 return;
-- 
1.7.4.1

